app/controllers/trainees_controller.rb

```diff
@@ -1,3 +1,38 @@
 class TraineesController < ApplicationController
-  def show; end
+  def show
+    @trainee = Trainee.find(params[:id])
+  end
+
+  def new
+    @trainee = Trainee.new
+  end
+
+  def create
+    trainee = Trainee.create(trainee_params)
+    redirect_to trainee_path(trainee)
+  end
+
+  def personal_details
```

Comment:
Is there any reason why we can't just have something like:

```ruby
  resources :trainees, only: %i[show new create update] do
    member do
      get "personal-details", to: "personal_details#idex"
    end
  end
```

So that we use a separate class for detailing with the concerns of that route and avoid sprinkling custom actions?

Reply:
I vote for this.

I was thinking the same thing during tech breakdown, but the conversation veered towards having a single update action that could handle update from any of these forms. But it does feel like each of these sections is going to be a separate entity of sorts, so I'd be inclined to split these out into their own controllers.

Would it makes sense to have these in a subdirectory, though? `trainees/personal_details` maybe?

Reply:
done. i've nested to `trainees`